payjoin-cli/src/app/config.rs

```diff
@@ -58,55 +58,53 @@ pub struct Config {
 }
 
 impl Config {
-    /// Version flags in order of precedence (newest to oldest)
-    const VERSION_FLAGS: &'static [(&'static str, u8)] = &[("bip77", 2), ("bip78", 1)];
-
     /// Check for multiple version flags and return the highest precedence version
-    fn determine_version(matches: &ArgMatches) -> Result<u8, ConfigError> {
+    fn determine_version(cli: &Cli) -> Result<u8, ConfigError> {
         let mut selected_version = None;
-        for &(flag, version) in Self::VERSION_FLAGS.iter() {
-            if matches.get_flag(flag) {
-                if selected_version.is_some() {
-                    return Err(ConfigError::Message(format!(
-                        "Multiple version flags specified. Please use only one of: {}",
-                        Self::VERSION_FLAGS
-                            .iter()
-                            .map(|(flag, _)| format!("--{flag}"))
-                            .collect::<Vec<_>>()
-                            .join(", ")
-                    )));
-                }
-                selected_version = Some(version);
+
+        // Check for BIP77 (v2)
+        if cli.flags.bip77.unwrap_or(false) {
+            selected_version = Some(2);
+        }
+
+        // Check for BIP78 (v1)
+        if cli.flags.bip78.unwrap_or(false) {
+            if selected_version.is_some() {
+                return Err(ConfigError::Message(
+                    "Multiple version flags specified. Please use only one of: --bip77, --bip78"
```

Comment:
I aimed to not change any language or behavior here, but the difference between the `v1`/`v2` features and `bip77`/`bip78` would be helpful. _Is_ `bip77` a "version" flag? If not, I can update the language in another PR.